### ISSUE TITLE
Add support for cassandra

### ISSUE THREAD

**Issue by dawidmalina** [naver/pinpoint#789]
It would be.nice to have dedicated profiler for cassandra same as we have for redis.

**Comment by youtome** [comment]
+1

**Comment by emeroad** [comment]
We have a plan to support hbase and cassandra.

**Comment by dawidmalina** [comment]
:thumbsup:

**Comment by dawidmalina** [comment]
Is there any updates for this?

**Comment by dawidmalina** [comment]
I found this: https://github.com/dynaTrace/Dynatrace-Apache-Cassandra-Fastpack

Based on this I've created a list with classes and methods that (probably) should be traced. I hope it will help.

Class Name | Method Name
------------ | -------------
org.apache.cassandra.service.AbstractWriteResponseHandler | get
org.apache.cassandra.thrift.CassandraServer | getSlice, state
org.apache.cassandra.db.compaction.CompactionManager | doCompaction
org.apache.cassandra.thrift.Cassandra$Processor$ProcessFunction | process
org.apache.cassandra.thrift.Cassandra$Processor | process
org.apache.cassandra.service.StorageProxy | mutate, performWrite, insertLocal, mutateCounter, applyCounterMutationOnLeader, applyCounterMutationOnCoordinator, read, scan
org.apache.cassandra.db.compaction.CompactionTask | execute
org.apache.cassandra.db.compaction.CompactionManager$CompactionExecutorStatsCollector | beginCompaction
me.prettyprint.cassandra.model.ExecutingKeyspace | doExecute
org.apache.cassandra.thrift.Cassandra$Iface | get, get_slice, get_count, multiget_slice, multiget_count, get_range_slices, get_indexed_slices, insert,  remove, remove_counter, batch_mutate
com.datastax.tutorial.InsertRowsForColumnFamilies$ExecTask | run

**Comment by Xylus** [comment]
For what it's worth, the Thrift generated classes (at least the client side) should already be handled by the Thrift plugin
```
org.apache.cassandra.thrift.Cassandra$Iface (client)
org.apache.cassandra.thrift.Cassandra$Processor$ProcessFunction (server)
org.apache.cassandra.thrift.Cassandra$Processor (server)
```
A proper Cassandra support would be the way to go eventually, and this should be in the works (not 100% sure on this)

**Comment by dawidmalina** [comment]
This also should be helpful : https://github.com/spring-projects/spring-insight-plugins/tree/master/collection-plugins/cassandra

**Comment by Xylus** [comment]
Merged via #1643 
Possible future additions - bind variable tracing.
